dvc/remote/base.py

```diff
@@ -820,9 +826,102 @@ def cache_exists(self, checksums, jobs=None, name=None):
         Returns:
             A list with checksums that were found in the remote
         """
-        if not self.no_traverse:
-            return list(set(checksums) & set(self.all()))
 
+        if self.no_traverse:
+            return self._cache_exists_no_traverse(checksums, jobs, name)
+
+        # Fetch one parent cache dir for estimating size of entire remote cache
+        checksums = frozenset(checksums)
+        prefix = "0" * self.TRAVERSE_PREFIX_LEN
+        traverse_dirs = pow(16, self.TRAVERSE_PREFIX_LEN)
```

Comment:
could you briefly describe what's the logic here - why do we translate this into remote size if prefix is empty

Reply:
I wrote this with the assumption that `TRAVERSE_PREFIX_LEN` would never be zero - the remotes which don't use the prefix at all (ssh, local) have their own overridden `cache_exists()` methods. But it does make sense to account for an empty prefix here as well.

Reply:
sorry, my question was confusing: `prefix empty == size of the object with that prefix in the if below` ... I was asking mostly about the `if` below when we already checked the size of the remote at the prefix.

Reply:
If the fetched prefix is empty, we still need to estimate that the remote size is `< 256` or `< 4096` files (depending on prefix length) rather than `0`. Later on we use that size to estimate how many pages (`size / 1000` for most remotes) the full list of remote objects would return, and we want a non-zero page count.

Reply:
I think the name confused me a bit ... since it's not so much about `dirs` in general case ... `total_prefixes`?

Reply:
renamed var to `total_prefixes` in 65d7e22

Reply:
thanks 🙏